# Topical Authority Finding

In [34]:
import pandas as pd
import numpy as np
import nltk
from nltk.corpus import stopwords
import operator
import os
print(os.getcwd())
print(os.listdir('/Users/Brandon/Yelp Dataset/dataset'))
import json
import glob

/Users/Brandon/Yelp Dataset/Yelp-FA17/local-elites/brandon
['.DS_Store', 'business.json', 'checkin.json', 'photos.json', 'review.json', 'review1.json', 'review2.json', 'review3.json', 'review4.json', 'review5.json', 'review5000.json', 'tip.json', 'user.json']


In [60]:
def load_json_to_df(datapass):
    '''
    Load the json file and parse the file to pandas dataframe format
    
    Input:
        datapass(str) : directory to the json file
    Output:
        df(dataframe) : pandas dataframe object
    '''
    
    data = [] 
    with open(datapass) as data_file: 
        for f in data_file:
            data.append(json.loads(f))
    df = pd.DataFrame(data)
    return df

In [ ]:
import_file = "/Users/Brandon/Yelp Dataset/dataset/review.json"
#json_pattern = os.path.join(import_file,'review[0-9].json')
#file_list = glob.glob(json_pattern)

review = load_json_to_df(import_file)

In [59]:
import_file = '/Users/Brandon/Yelp Dataset/dataset/business.json'
business = pd.read_json(import_file, lines=True)

In [74]:
%%time
import_file = '/Users/Brandon/Yelp Dataset/dataset/user5000.json'
user = pd.read_json(import_file, lines=True)

CPU times: user 334 ms, sys: 752 ms, total: 1.09 s
Wall time: 2.38 s


In [196]:
list(user)

['average_stars',
 'compliment_cool',
 'compliment_cute',
 'compliment_funny',
 'compliment_hot',
 'compliment_list',
 'compliment_more',
 'compliment_note',
 'compliment_photos',
 'compliment_plain',
 'compliment_profile',
 'compliment_writer',
 'cool',
 'elite',
 'fans',
 'friends',
 'funny',
 'name',
 'review_count',
 'useful',
 'user_id',
 'yelping_since']

Find which businesses are classified as a specific category. Here we do an example just looking at restaurants.

In [156]:
#user[user['review_count']>5000]
list(business)
def categoryFind(x,cat):
    return cat in x;
restaurantIndex = business['categories'].apply(categoryFind,cat='Restaurants')
restaurants = business[restaurantIndex]
len(restaurants)


51613

We have 51613 businesses, let's try to look at just the first 500 for now.

Let's take a look at reviews

In [191]:
#list(review)
list(business)

['address',
 'attributes',
 'business_id',
 'categories',
 'city',
 'hours',
 'is_open',
 'latitude',
 'longitude',
 'name',
 'neighborhood',
 'postal_code',
 'review_count',
 'stars',
 'state']

In [184]:
%%time
businessReviews = pd.merge(review, business,on='business_id',right_index=True,sort=False)

CPU times: user 9.26 s, sys: 1.74 s, total: 11 s
Wall time: 11.2 s


In [195]:
businessReviews#[['business_id','user_id']]
#list(businessReviews)

business_id  cool        date  funny  \
0        uYHaNptLzDLoV_JZ_MuzUA     0  2016-07-12      0   
1        uYHaNptLzDLoV_JZ_MuzUA     0  2016-10-02      0   
2        uYHaNptLzDLoV_JZ_MuzUA     0  2015-09-17      0   
3        uYHaNptLzDLoV_JZ_MuzUA     0  2016-08-21      0   
4        uYHaNptLzDLoV_JZ_MuzUA     0  2013-11-20      0   
5        uYHaNptLzDLoV_JZ_MuzUA     0  2016-06-05      0   
6        uYHaNptLzDLoV_JZ_MuzUA     0  2015-02-21      0   
7        uYHaNptLzDLoV_JZ_MuzUA     0  2013-07-07      0   
8        uYHaNptLzDLoV_JZ_MuzUA     2  2013-04-27      0   
9        uYHaNptLzDLoV_JZ_MuzUA     0  2015-04-13      0   
10       uYHaNptLzDLoV_JZ_MuzUA     1  2016-11-08      1   
11       uYHaNptLzDLoV_JZ_MuzUA     0  2015-07-27      0   
12       uYHaNptLzDLoV_JZ_MuzUA     0  2014-05-07      0   
13       uYHaNptLzDLoV_JZ_MuzUA     0  2015-02-26      0   
14       uYHaNptLzDLoV_JZ_MuzUA     1  2015-08-21      0   
15       uYHaNptLzDLoV_JZ_MuzUA     0  2013-12-07      0   
16       jQsNFOzDpxPmOurSWCg1vQ     0  2017-06-03      0   
17       jQsNFOzDpxPmOurSWCg1vQ     0  2015-03-26      0   
18       jQsNFOzDpxPmOurSWCg1vQ     1  2012-12-30      1   
19       jQsNFOzDpxPmOurSWCg1vQ     1  2009-01-12      1   
20       jQsNFOzDpxPmOurSWCg1vQ     0  2015-07-11      0   
21       jQsNFOzDpxPmOurSWCg1vQ     0  2015-05-27      0   
22       jQsNFOzDpxPmOurSWCg1vQ     1  2015-02-28      1   
23       jQsNFOzDpxPmOurSWCg1vQ     0  2010-04-05      0   
24       jQsNFOzDpxPmOurSWCg1vQ     0  2015-05-22      0   
25       jQsNFOzDpxPmOurSWCg1vQ     0  2011-06-15      0   
26       jQsNFOzDpxPmOurSWCg1vQ     0  2017-03-12      0   
27       jQsNFOzDpxPmOurSWCg1vQ     1  2016-12-19      0   
28       jQsNFOzDpxPmOurSWCg1vQ     0  2011-08-10      0   
29       jQsNFOzDpxPmOurSWCg1vQ     0  2013-06-17      0   
...                         ...   ...         ...    ...   
4736867  iJ4uoZkPNJ39Vfh0HirI3w     0  2016-04-04      0   
4736868  iJ4uoZkPNJ39Vfh0HirI3w     0  2015-04-28      0   
4736869  iJ4uoZkPNJ39Vfh0HirI3w     0  2015-10-02      0   
4736870  iJ4uoZkPNJ39Vfh0HirI3w     1  2016-08-08      0   
4736871  iJ4uoZkPNJ39Vfh0HirI3w     0  2015-05-30      0   
4736872  iJ4uoZkPNJ39Vfh0HirI3w     1  2016-09-12      1   
4736873  iJ4uoZkPNJ39Vfh0HirI3w     0  2016-05-31      0   
4736874  iJ4uoZkPNJ39Vfh0HirI3w     0  2015-04-13      0   
4736875  iJ4uoZkPNJ39Vfh0HirI3w     1  2016-08-14      0   
4736876  iJ4uoZkPNJ39Vfh0HirI3w     0  2016-09-18      0   
4736877  iJ4uoZkPNJ39Vfh0HirI3w     0  2016-05-21      1   
4736878  iJ4uoZkPNJ39Vfh0HirI3w     0  2016-02-02      0   
4736879  iJ4uoZkPNJ39Vfh0HirI3w     6  2015-08-07      8   
4736880  iJ4uoZkPNJ39Vfh0HirI3w     0  2016-09-17      0   
4736881  iJ4uoZkPNJ39Vfh0HirI3w     0  2016-06-25      0   
4736882  iJ4uoZkPNJ39Vfh0HirI3w     1  2015-09-19      0   
4736883  iJ4uoZkPNJ39Vfh0HirI3w     0  2016-03-20      0   
4736884  iJ4uoZkPNJ39Vfh0HirI3w     2  2015-06-29      1   
4736885  iJ4uoZkPNJ39Vfh0HirI3w     0  2015-09-11      0   
4736886  iJ4uoZkPNJ39Vfh0HirI3w     1  2015-08-30      1   
4736887  iJ4uoZkPNJ39Vfh0HirI3w     0  2016-01-03      0   
4736888  iJ4uoZkPNJ39Vfh0HirI3w     0  2015-09-14      0   
4736889  iJ4uoZkPNJ39Vfh0HirI3w     0  2016-02-18      0   
4736890  iJ4uoZkPNJ39Vfh0HirI3w     0  2016-01-25      0   
4736891  iJ4uoZkPNJ39Vfh0HirI3w     0  2015-09-14      0   
4736892  iJ4uoZkPNJ39Vfh0HirI3w     2  2015-07-19      3   
4736893  iJ4uoZkPNJ39Vfh0HirI3w     2  2015-09-07      1   
4736894  iJ4uoZkPNJ39Vfh0HirI3w     0  2015-09-10      2   
4736895  iJ4uoZkPNJ39Vfh0HirI3w     0  2015-04-21      0   
4736896  iJ4uoZkPNJ39Vfh0HirI3w     0  2016-06-24      0   

                      review_id  stars_x  \
0        VfBHSwC5Vz_pbFluy07i9Q        5   
1        3zRpneRKDsOPq92tq7ybAA        3   
2        ne5WhI1jUFOcRn-b-gAzHA        3   
3        llmdwOgDReucVoWEry61Lw        4   
4        DuffS87NaSMDmIfluvT83g        5   
5        GvLmUkjUrOyFH8KFnmT1uw    